app/models/payment.rb

```diff
@@ -221,6 +217,24 @@ def convert_old_status!
     save!
   end
 
+  # Manually mark that a payment was received.
+  def mark_received
+    status_received!
+  end
+
+  # In transaction,
+  # mark payment received and ticket request completed
+  # Send off PaymentMailer for payment received
+  def request_completed
+    mark_received
```

Comment:
Do we want to wrap these two in a transaction?

Reply:
I'll move the transaction from the controller.